### PR TITLE
ensure running `powershell` within PowerShell starts instance of currently running PowerShell

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
modify PATH env var at startup so that $PSHOME is in front

Fix https://github.com/PowerShell/PowerShell/issues/4194

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 1]
```diff
@@ -134,6 +134,18 @@ internal sealed partial class ConsoleHost
             }
 #endif
 
+            // put PSHOME in front of PATH so that calling `powershell` within `powershell` always starts the same running version
+            StringBuilder path = new StringBuilder(Environment.GetEnvironmentVariable("PATH"));
+            string pathSeparator = ";";
+            if (!Platform.IsWindows)
+            {
+                pathSeparator = ":";
+            }
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: You can use `Path.PathSeparator`.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 2]
```diff
@@ -134,6 +134,18 @@ internal sealed partial class ConsoleHost
...
+            string pshome = Utils.DefaultPowerShellAppBase;
+            path.Replace(pathSeparator + pshome, "");
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: Why is this needed?

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 3]
```diff
@@ -134,6 +134,18 @@ internal sealed partial class ConsoleHost
...
+            path.Insert(0, pshome + pathSeparator);
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: We should check if the first path in `$env:PATH` is `pshome`, and only insert if it's not. Otherwise, the nested powershell instance will have 2 `pshome` paths in `$env:PATH`

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs (hunk 1 above)
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: ok

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs (hunk 2 above)
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: I thought that the PWD got appended to the end of PATH so I didn't want it on both ends, looks like that isn't the case.  I'll remove.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs (hunk 3 above)
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: ok

**Review by daxian-dbw — APPROVED** [review]
LGTM

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 4]
```diff
@@ -134,6 +134,14 @@ internal sealed partial class ConsoleHost
             }
 #endif
 
+            // put PSHOME in front of PATH so that calling `powershell` within `powershell` always starts the same running version
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: 1. Is this right for other applications?
2. I would prefer more deterministic `[System.Diagnostics.Process]::GetCurrentProcess().MainModule.FileName`
